federation/sssd/src/main/java/org/keycloak/federation/sssd/api/Sssd.java

```diff
@@ -139,32 +140,10 @@ public String getLastName() {
             return lastName;
         }
 
-        @Override
-        public boolean equals(Object o) {
-            if (o == null) return false;
-
-            UserModel userModel = (UserModel) o;
-            if (firstName != null && !firstName.equals(userModel.getFirstName())) {
-                return false;
-            }
-            if (lastName != null && !lastName.equals(userModel.getLastName())) {
-                return false;
-            }
-            if (email != null) {
-                return email.equals(userModel.getEmail());
-            }
-            if (email != userModel.getEmail()) {
-                return false;
-            }
-            return true;
-        }
-
-        @Override
-        public int hashCode() {
-            int result = email != null ? email.hashCode() : 0;
-            result = 31 * result + (firstName != null ? firstName.hashCode() : 0);
-            result = 31 * result + (lastName != null ? lastName.hashCode() : 0);
-            return result;
+        public boolean equivalentTo(UserModel user) {
```

Comment:
Never remove `equals` and `hashcode` implementation

Reply:
I removed it because it does not follow the contract of how .hashCode()/.equals() supposed to work as it compares incompatible types and it breaks a.equals(b) == b.equals(a)